packages/flutter_tools/lib/src/commands/build_macos_framework.dart

```diff
@@ -0,0 +1,305 @@
+import '../runner/flutter_command.dart' show DevelopmentArtifact, FlutterCommandResult;
+import '../version.dart';
+import 'build_ios_framework.dart';
+
+/// Produces a .framework for integration into a host macOS app. The .framework
+/// contains the Flutter engine and framework code as well as plugins. It can
+/// be integrated into plain Xcode projects without using or other package
+/// managers.
+class BuildMacOSFrameworkCommand extends BuildFrameworkCommand {
+  BuildMacOSFrameworkCommand({
+    super.flutterVersion,
+    required super.buildSystem,
+    required super.verboseHelp,
+    super.cache,
+    super.platform,
+  });
+
+  @override
+  final String name = 'macos-framework';
+
+  @override
+  final String description = 'Produces .xcframeworks for a Flutter project '
+      'and its plugins for integration into existing, plain macOS Xcode projects.\n'
+      'This can only be run on macOS hosts.';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{
+        DevelopmentArtifact.macOS,
+      };
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    final String outputArgument =
+        stringArg('output') ?? globals.fs.path.join(globals.fs.currentDirectory.path, 'build', 'macos', 'framework');
+
+    if (outputArgument.isEmpty) {
+      throwToolExit('--output is required.');
+    }
+
+    if (!project.macos.existsSync()) {
+      throwToolExit('Project does not support macOS');
+    }
+
+    final Directory outputDirectory =
+        globals.fs.directory(globals.fs.path.absolute(globals.fs.path.normalize(outputArgument)));
+    final List<BuildInfo> buildInfos = await getBuildInfos();
+    displayNullSafetyMode(buildInfos.first);
+    for (final BuildInfo buildInfo in buildInfos) {
+      globals.printStatus('Building macOS frameworks in ${getNameForBuildMode(buildInfo.mode)} mode...');
+      final String xcodeBuildConfiguration = sentenceCase(getNameForBuildMode(buildInfo.mode));
+      final Directory modeDirectory = outputDirectory.childDirectory(xcodeBuildConfiguration);
+
+      if (modeDirectory.existsSync()) {
+        modeDirectory.deleteSync(recursive: true);
+      }
+
+      if (boolArg('cocoapods') ?? false) {
+        produceFlutterPodspec(buildInfo.mode, modeDirectory, force: boolArg('force') ?? false);
+      }
+
+      // Build aot, create App.framework and copy FlutterMacOS.framework. Make XCFrameworks.
+      await _produceAppFramework(buildInfo, modeDirectory);
+
+      // Build and copy plugins.
+      final Directory buildOutput = modeDirectory.childDirectory('macos');
+      await processPodsIfNeeded(project.macos, getMacOSBuildDirectory(), buildInfo.mode);
+      if (hasPlugins(project)) {
+        await _producePlugins(xcodeBuildConfiguration, buildOutput, modeDirectory);
+      }
+
+      final Status status =
+          globals.logger.startProgress(' └─Moving to ${globals.fs.path.relative(modeDirectory.path)}');
```

Comment:
This doesn't look like it matches what the following try block is doing?

Reply:
You're right, I'll change to a `printStatus` since I still want to convey the path.